### DISCUSSION TITLE
Running linux perf on Mariner

### DISCUSSION THREAD

**Discussion by wmcleish-msft** [microsoft/azurelinux#6476] [Q&A] [answered]
Is it possible to run linux perf on mariner? I can't seem to find it in the available packages in tdnf and I'm having difficulties trying to build it from source on my Mariner VM. Maybe I'm missing something since I haven't had to use perf before. If it is not available on Mariner then what profiling tools are? Any help or suggestions would be greatly appreciated. Thank you!

**Comment by mfrw** [comment, marked as answer]
Hello,

`perf` is provided by the package `kernel-tools`.

Use the following snippet to install perf:
```bash
sudo tdnf install kernel-tools
```

Also, Mariner ships with [bpftrace](https://github.com/iovisor/bpftrace) & [bcc](https://github.com/iovisor/bcc) as well, which are tools that leverage eBPF.

**Reply by wmcleish-msft** [reply]
Thank you for this!
